artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/cursor/impl/PageCursorProviderImpl.java

```diff
@@ -517,6 +528,23 @@ protected void cleanupComplete(ArrayList<PageSubscription> cursorList) throws Ex
 
       storeBookmark(cursorList, currentPage);
 
+      // notice this will be using the OperationContext started by cleanup
+      storageManager.afterCompleteOperations(new IOCallback() {
+         @Override
+         public void done() {
+
+            for (PageSubscription cursor : cursorList) {
+               cursor.enableAutoCleanup();
+            }
+         }
+
+         @Override
+         public void onError(int errorCode, String errorMessage) {
```

Comment:
what about error handling?

Reply:
some place else would be already listening for it and the broker would shutdown on critical error.